src/ui/ui.rs

```diff
@@ -390,3 +435,45 @@ fn event_to_nvim_input(e: &gdk::EventKey) -> Option<String> {
 
     Some(format!("<{}>", input))
 }
+
+fn handle_drop_uri_list(nvim: GioNeovim, data: &gtk::SelectionData) {
+    let filenames = data.get_uris();
+
+    spawn_local(async move {
+        for filename in filenames {
+            if let Ok(filename) = urlencoding::decode(filename.as_str()) {
+                // errors from `command` are nvim command errors and will be
+                // shown to the user except for E37, so we ignore handling them here
+                //
+                // first try to edit the file, if failed due to the current buffer
+                // is modified, edit the file in a new split
+                if let Err(e) = nvim.command(&format!("e {}", filename)).await {
+                    if let nvim_rs::error::CallError::NeovimError(_, s) =
+                        e.as_ref()
+                    {
+                        // only respond to
+                        // "Vim(edit):E37: No write since last change (add ! to override)"
+                        if s.starts_with("Vim(edit):E37") {
+                            // ignore errors
+                            nvim.command(&format!("sp {}", filename))
+                                .await
+                                .ok();
+                        }
+                    }
+                }
```

Comment:
In the GVim code, it determines if it needs to `split` or `edit` by checking if the current buffer has `changed`, but I could not find a way to check for that, even `nvim_get_option('modified')` does not provide that